crates/sui-core/src/authority_server.rs

```diff
@@ -379,6 +382,21 @@ impl ValidatorService {
             .data()
             .validity_check(epoch_store.protocol_config())?;
 
+        match state.get_execution_state().await {
```

Comment:
Putting the check here seems indicate that we can still sign txn. Is this intended?

Reply:
Yes. But I went back and forth on this. My view is that tx signing in the event of an execution nondeterminism, for exmple, should be safe, given that the split brain could not have occurred if there was an issue in the transaction signing layer (we would never get that far). And it makes sense to me to have as much "liveness" as we can safely allow in such a case.

Also, if it were between either blocking at signing stage or execution stage, we definitely would want to block at execution stage, since a user could sit on a signed transaction, then we fork, then they could execute.

But I could definitely be convinced that maybe we should deny both.